src/txn/txn_ckpt.c

```diff
@@ -634,12 +634,14 @@ __checkpoint_prepare(WT_SESSION_IMPL *session, bool *trackingp, const char *cfg[
      *
      * Using a transaction id for prepared rollback that is allocated after the second checkpoint
      * snapshot has issues when stale reserved transaction id used for the history store updates and
-     * the data store page is skipped in the checkpoint.
+     * the data store page is skipped in the checkpoint until all the pages that have restored
+     * prepared updates are forced to be checkpointed in the current checkpoint.
      *
      * The checkpoint snapshot max can also be used for this purpose, instead of allocating a new
-     * reserved transaction id. It is better to use a dedicated transaction id as the checkpoint
-     * snapshot max is allocated to a session and used for other operations can lead to confusion
-     * when an issue occurs.
+     * reserved transaction id. It has the same problem of forcing all the pages with restored
```

Comment:
Why you say problem? Do you mean it can solve the problem here?

Reply:
I updated the comment to have a better understanding of alternatvies.